### ISSUE TITLE
if let true = k > 1 {} and if let true = true {}

### ISSUE THREAD

**Issue by Fundevoge** [rust-lang/rust-clippy#11917]
### What it does

Warn you that this is bad style, recommend removing the let binding

### Advantage

Clearly, it is not idiomatic to use refutable pattern matching on booleans

### Drawbacks

None as far as I know

### Example

```rust
let k = 5;
if let true = k > 1 { /* */ }
```

Could be written as:

```rust
let k = 5;
if k > 1 { /* */ }
```

AND

```rust
if let true = true { /* some code */ }
```

Could be written as:

```rust
/* some code */
```

AND

```rust
if let true = false { /* some code */ }
```

Should trigger an unreachable code warning

**Comment by Fundevoge** [comment]
I agree, just simplifying `if let true = some_condition` to `if some_condition` and `if let false = some_condition` to `if !some_condition` (or `if !(some_condition)`, depending on whether parentheses are needed) should be enough for other rules to kick in
